src/test/java/teammates/test/cases/logic/CoursesLogicTest.java

```diff
@@ -438,12 +438,22 @@ private void testGetCoursesForStudentAccount() throws Exception {
         assertEquals(2, courseList.size());
 
         CourseAttributes course1 = dataBundle.courses.get("typicalCourse1");
-        assertEquals(course1.getId(), courseList.get(0).getId());
-        assertEquals(course1.getName(), courseList.get(0).getName());
 
         CourseAttributes course2 = dataBundle.courses.get("typicalCourse2");
-        assertEquals(course2.getId(), courseList.get(1).getId());
-        assertEquals(course2.getName(), courseList.get(1).getName());
+
+        boolean isCourse1Present = false;
+        boolean isCourse2Present = false;
+
+        for (CourseAttributes course : courseList) {
+            if (course.compareTo(course1) == 0) {
```

Comment:
`compareTo` does not replace `equals`. Take this analogy: when you're sorting a deck of poker cards numerically (disregarding the suits), a 5 of hearts and a 5 of diamonds will have the same value, but they are not the same card.
If `equals` is not defined for this class (seems like it is the case), just use the way it is done before, i.e. comparing the ID and the name.